src/expr/src/relation/func.rs

```diff
@@ -749,6 +749,64 @@ fn generate_series_int64(start: Datum, stop: Datum) -> impl Iterator<Item = (Row
     (start..=stop).map(move |i| (Row::pack_slice(&[Datum::Int64(i)]), 1))
 }
 
+fn generate_series_step_int32(
```

Comment:
nbd, but if you're up for it, genericizing this function would be cool because the logic is identical. You'd have to unwrap the datums before calling the function, but I think everything else could work. Would also give you the `numeric` impl for free, I think. No sweat and not blocking, though.